pkgs/applications/editors/emacs/site-start.el

```diff
@@ -53,6 +53,9 @@ least specific (the system profile)"
                         (nix--profile-paths))
                 native-comp-eln-load-path)))
 
+;;; Disable asynchronously deferred native-comp by default
+(setq native-comp-deferred-compilation nil)
```

Comment:
Why do we want this? Wouldn't it have performance benefits to leave this enabled?

Reply:
When someone has lots of packages from ELPA, then after launching Emacs it will try to compile all of these packages which is quite heavy on CPU. I agree with the author of the PR that it is better to make the native compilation for user packages opt-in.

Reply:
Deleting this line `(setq native-comp-deferred-compilation nil)` does have performance benifits but users will need some compilation if they install packages not from nixpkgs. (You can see the release notes for more information about this).

Some users don't want that compilation:

@wahjava wrote in https://github.com/NixOS/nixpkgs/pull/168076#issuecomment-1094511748
> it uses 100% CPU (in its default configuration) when you install any new packages for native compilation

I set `native-comp-deferred-compilation` to `nil` in this pr to have less impact on users. However, I would love to remove this line `(setq native-comp-deferred-compilation nil)` and instead add instructions that users don't want this can set `(setq native-comp-deferred-compilation nil)` in their init file, like what @axelf4 said and @adisbladis agreed in #168076:

> Would not those users be better off disabling just-in-time native compilation in their init files, while still benefiting from the packaged Emacs distribution being ahead-of-time compiled?

What do you think?

Reply:
I think I prefer not setting this. Everything else in this file looks like me to distro-compatibility code, which I think is a lot more appropriate for us to decide on, as opposed to user preferences, which I think should be left to upstream (for defaults) and, well, the users :)

I think removing it but suggesting it to users in the release notes would be good.

Reply:
Regarding upstream's default, one may argue that upstream does not enable native-comp by default. However, since this change is mentioned in the release notes, it should be fine.

@lheckemann What you suggested has been done.